### PR TITLE
feat: injectable

### PR DESCRIPTION
**Pull request by vladimir-hodaryonok-idf**

### CODE REVIEW

File: lib/di/injector.dart [hunk 1]
```diff
@@ -1,7 +1,10 @@
 import 'package:domain/domain.dart';
+import 'package:get_it/get_it.dart';
 import 'package:presentation/presentation.dart';
 
-Future<void> initInjector() async {
-  initPresentationModule();
-  initDomainModule();
+final getIt = GetIt.instance;
+
+Future<void> configureDependencies() async {
```
Comment (egor-slatvinskiy-idf) [inline lib/di/injector.dart]: why do you need an asynchronous method?

File: pubspec.yaml [hunk 2]
```diff
@@ -14,6 +14,8 @@ dependencies:
   injectable: ^1.5.3
   presentation:
     path: packages/presentation
+  domain:
```
Comment (egor-slatvinskiy-idf) [inline pubspec.yaml]: you don't need it here

File: packages/presentation/pubspec.yaml [hunk 3]
```diff
@@ -12,8 +12,11 @@ dependencies:
   flutter:
     sdk: flutter
   get_it: ^7.2.0
```
Comment (egor-slatvinskiy-idf) [inline packages/presentation/pubspec.yaml]: you don't need it here

File: pubspec.yaml [hunk 4]
```diff
@@ -10,9 +10,8 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-
-
-  cupertino_icons: ^1.0.2
+  get_it: ^7.2.0
```
Comment (Egor-Slatvinskiy-idf) [inline pubspec.yaml:13]: -//-

File: pubspec.yaml [hunk 5]
```diff
@@ -10,9 +10,8 @@ environment:
...
+  injectable: ^1.5.3
```
Comment (egor-slatvinskiy-idf) [inline pubspec.yaml]: -//-

File: pubspec.yaml [hunk 6]
```diff
@@ -23,6 +22,8 @@ dev_dependencies:
     sdk: flutter
 
   flutter_lints: ^2.0.0
+  build_runner: 2.2.0
+  injectable_generator: 1.5.4
```
Comment (egor-slatvinskiy-idf) [inline pubspec.yaml]: -//-

**Review by egor-slatvinskiy-idf — APPROVED** [review]

**Review by elizaveta-seleznyova-idf — APPROVED** [review]

File: packages/domain/lib/src/di/domain_injector.config.dart [hunk 7]
```diff
@@ -0,0 +1,24 @@
+// GENERATED CODE - DO NOT MODIFY BY HAND
+
+// **************************************************************************
+// InjectableConfigGenerator
+// **************************************************************************
+
+// ignore_for_file: no_leading_underscores_for_library_prefixes
+import 'package:get_it/get_it.dart' as _i1;
+import 'package:injectable/injectable.dart' as _i2;
```
Comment (alexei-kruk-idf) [inline packages/domain/lib/src/di/domain_injector.config.dart:9]: gitignore

File: packages/presentation/lib/src/di/presentation_injector.config.dart [hunk 8]
```diff
@@ -0,0 +1,32 @@
+// GENERATED CODE - DO NOT MODIFY BY HAND
+
+// **************************************************************************
+// InjectableConfigGenerator
+// **************************************************************************
+
+// ignore_for_file: no_leading_underscores_for_library_prefixes
+import 'package:domain/domain.dart' as _i5;
+import 'package:flutter/material.dart' as _i3;
+import 'package:get_it/get_it.dart' as _i1;
```
Comment (alexei-kruk-idf) [inline packages/presentation/lib/src/di/presentation_injector.config.dart:10]: -//-

**Review by alexei-kruk-idf — APPROVED** [review]

**Pull request merged**